vendor/sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client/client.go

```diff
@@ -114,6 +120,10 @@ func (t *grpcTunnel) serve(c clientConn) {
 					err:    resp.Error,
 					connid: resp.ConnectID,
 				}
+				// We should only get DialResp once. If we get the DialResp
+				// twice, we will get an exception trying to write to a closed
+				// channel.
+				close(ch)
```

Comment:
what's the failure mode if this happens?

Reply:
When we get the second dial response, before we've had a chance to remove the channel from the pendingDial map (talking milliseconds here) then line 119 will get an error writing to a closed channel. That should leave us with a stacktrace and kill this goroutine. That should stop also stop us from hanging the GRPC tunnel.

Reply:
That will panic and crash the server, right?

Reply:
Last time I checked the answer would have been no but apparently we've changed that default since last time I checked.

Reply:
spoke with walter offline, I think the plan is to change from a `close(ch)` (which can panic the server) to something like this:

```go
select {
case ch <- dialResult{err: resp.Error, connid: resp.ConnectID}:
  // write to channel succeeded
default:
  // write to channel blocked, which is unexpected
  ... log nasty message ...
  return
}
```

Reply:
Resolved with https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/221 (Post v0.0.18)